google/cloud/bigtable/internal/data_connection_impl.cc

```diff
@@ -71,6 +72,24 @@ bigtable::RowReader DataConnectionImpl::ReadRows(
   return MakeRowReader(std::move(impl));
 }
 
+std::vector<bigtable::FailedMutation> DataConnectionImpl::BulkApply(
+    std::string const& app_profile_id, std::string const& table_name,
+    bigtable::BulkMutation mut) {
+  if (mut.empty()) return {};
+  bigtable::internal::BulkMutator mutator(
+      app_profile_id, table_name, *idempotency_policy(), std::move(mut));
+  auto retry = retry_policy();
```

Comment:
nit: this always allocate a retry and backoff policy, even though on success (the common case?) they are not used.  Most likely not important, but if you are looking for micro-optimizations, there is one.

Reply:
Done.